### PR TITLE
Mabel 0.1.1 (new formula)

### PR DESCRIPTION
**Pull request by FIGBERT**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/mabel.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Mabel < Formula
+  desc "Fancy BitTorrent client for the terminal"
+  homepage "https://github.com/smmr-software/mabel"
+  url "https://github.com/smmr-software/mabel.git",
+    tag:      "v0.1.1",
+    revision: "23490e76aea68357b224dd4d7f1aed822ce2cd46"
```
Comment (carlocab) [inline Formula/mabel.rb:6]:
```suggestion
  url "https://github.com/smmr-software/mabel.git",
      tag:      "v0.1.1",
      revision: "23490e76aea68357b224dd4d7f1aed822ce2cd46"
```

File: Formula/mabel.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      -X main.builtBy=homebrew
+      -X main.date=#{time.iso8601}
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags)
+  end
+
+  test do
+    vrsn_out = shell_output("#{bin}/mabel --version")
+    assert_match "Mabel #{version}", vrsn_out
+    assert_match "Built by: homebrew", vrsn_out
```
Comment (carlocab) [inline Formula/mabel.rb:26]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

We can keep this test to make sure that `main.version` works, but we need an additional test to make sure what we installed is working properly.

File: Formula/mabel.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  head "https://github.com/smmr-software/mabel.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{Utils.git_head}
+      -X main.builtBy=homebrew
```
Comment (carlocab) [inline Formula/mabel.rb:17]:
```suggestion
      -X main.builtBy=#{tap.user}
```

File: Formula/mabel.rb (hunk 2 above)
Comment (carlocab) [inline Formula/mabel.rb:26]:
```suggestion
    assert_match "Built by: #{tap.user}", vrsn_out
```

**Comment by FIGBERT** [comment]
I'm sure how to do about fixing the error in CI – any pointers, @carlocab?

**Comment by chenrui333** [comment]
```
  An exception occurred within a child process:
    Minitest::Assertion: Expected /open\ test\.torrent:\ no\ such\ file\ or\ directory/ to match "2022/06/29 17:50:48 open /dev/tty: no such device or address\n".
```

**Comment by FIGBERT** [comment]
I saw that error, @chenrui333, just not sure what to do about it as we don't reference `/dev/tty` anywhere in the test.

File: Formula/mabel.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    system "go", "build", *std_go_args(ldflags: ldflags)
+  end
+
+  test do
+    vrsn_out = shell_output("#{bin}/mabel --version")
+    assert_match "Mabel #{version}", vrsn_out
+    assert_match "Built by: #{tap.user}", vrsn_out
+
+    trnt_out = shell_output("#{bin}/mabel 'test.torrent' 2>&1", 1)
+    assert_match "open test.torrent: no such file or directory", trnt_out
```
Comment (carlocab) [inline Formula/mabel.rb:29]:
```suggestion
    error_message = if OS.linux? && ENV["HOMEBREW_GITHUB_ACTIONS"].present?
      "open /dev/tty: no such device or address"
    else
      "open test.torrent: no such file or directory"
    end
    assert_match error_message, trnt_out
```

**Review by carlocab — COMMENTED** [review]
We don't have a tty on the Linux runner, so `mabel` produces a different error. This might help.

**Review by carlocab — APPROVED** [review]
LGTM, thanks!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@FIGBERT, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**